tests/not.test.ts

```diff
@@ -100,4 +100,22 @@ describe('not', () => {
       ).toBe('hello');
     });
   });
+
+  it('should correctly exclude unit types with the unit wildcard', () => {
+    expect(
+      match<{ str: string | null | undefined }>({ str: 'hello' })
+        .with({ str: not(__.empty) }, ({ str }) => {
```

Comment:
I think I quite like `.nullish` for the reason you stated earlier - its already part of the common nomenclature because of` `??`, and its completely library agnostic.